code/modules/unit_tests/strange_reagent.dm

```diff
@@ -0,0 +1,151 @@
+	target.setBruteLoss(damage, FALSE) // no point running health update logic here
+	target.setFireLoss(damage, TRUE) // since we do it here
+	update_amounts(target)
+	if(percent >= 1)
+		target.death()
+	return TRUE
+
+/datum/unit_test/strange_reagent/proc/test_damage_but_no_death(target_type)
+	var/mob/living/target = allocate_new_target(target_type)
+	if(!damage_target_to_percentage(target, 0.8))
+		return
+
+	var/health = target.get_organic_health()
+	strange_reagent.expose_mob(target, INGEST, 20)
+	TEST_ASSERT_EQUAL(health, target.get_organic_health(), "Strange Reagent healed a target type [target.type] that was not dead.")
+
+/datum/unit_test/strange_reagent/proc/test_death_no_damage(target_type)
+	var/mob/living/target = allocate_new_target(target_type)
+	target.death()
+	update_amounts(target)
+	strange_reagent.expose_mob(target, INGEST, amount_needed_to_revive)
+	TEST_ASSERT_NOTEQUAL(target.stat, DEAD, "Strange Reagent did not revive a dead target type [target.type].")
+
+/datum/unit_test/strange_reagent/proc/test_death_with_damage(target_type)
+	var/mob/living/target = allocate_new_target(target_type)
+	if(!damage_target_to_percentage(target, 0.8))
+		return
+
+	target.death()
+	update_amounts(target)
+	strange_reagent.expose_mob(target, INGEST, amount_needed_to_revive)
+	TEST_ASSERT_NOTEQUAL(target.stat, DEAD, "Strange Reagent did not revive a dead target type [target.type].")
+
+/datum/unit_test/strange_reagent/proc/test_death_with_damage_but_not_enough_reagent(target_type)
+	var/mob/living/target = allocate_new_target(target_type)
+	if(!damage_target_to_percentage(target, 1.2))
+		return
+
+	update_amounts(target)
+	strange_reagent.expose_mob(target, INGEST, amount_needed_to_revive - 1)
+	TEST_ASSERT_EQUAL(target.stat, DEAD, "Strange Reagent revived a dead target type [target.type] without enough reagent.")
+
+/datum/unit_test/strange_reagent/proc/test_death_with_full_heal(target_type)
+	var/mob/living/target = allocate_new_target(target_type)
+	if(!damage_target_to_percentage(target, 0.8))
+		return
+
+	target.death()
+	update_amounts(target)
+	strange_reagent.expose_mob(target, INGEST, amount_needed_to_full_heal)
+	TEST_ASSERT_EQUAL(target_max_health, target.get_organic_health(), "Strange Reagent did not fully heal a dead target type [target.type] with the expected amount.")
+
+/datum/unit_test/strange_reagent/proc/test_death_from_damage(target_type)
+	var/mob/living/target = allocate_new_target(target_type)
+	if(!damage_target_to_percentage(target, 1.6)) // under the 2x damage cap
+		return
+
+	update_amounts(target)
+	strange_reagent.expose_mob(target, INGEST, amount_needed_to_revive)
+	TEST_ASSERT_NOTEQUAL(target.stat, DEAD, "Strange Reagent did not revive a target type [target.type] who died from damage.")
+
+/datum/unit_test/strange_reagent/proc/test_death_from_too_much_damage(target_type)
+	var/mob/living/target = allocate_new_target(target_type)
+	if(!damage_target_to_percentage(target, 3.3)) // 10% over the 2x damage cap
```

Comment:
The 2x damage cap thing should be a variable from strange reagent so that we can edit that without having to edit the test